### DISCUSSION TITLE
engine hat - any conflicts?

### DISCUSSION THREAD

**Discussion by psm12** [hatlabs/discussions#34] [Q&A]
Hi, I'm considering ordering an ESP32 sailor hat with both engine hat and oled display.

Are there any GPIO or other conflicts between these and also the other interfaces, NMEA200, 1Wire, I2C etc. that i should be aware of before purchase? Your guidance would be much appreciated.

Many thanks
Paul

**Comment by mairas** [comment]
Hi there, 

The engine hat uses GPIOs 12-15 for digital input but all other interfaces remain available. Of course, if you use a lot of them, you might need to use a bigger enclosure to fit all the connectors you want (but that's already a consideration for the engine hat by itself).

**Reply by psm12** [reply]
Marias,
Thanks for the quick response.

Paul

**Comment by mairas** [comment]
Adding a bit: the Engine Hat does use the I2C bus for the analog input but the bus can be shared with other users.

**Reply by psm12** [reply]
Thanks Marias,
I should have just looked at the schematics more carefully!

**Reply by mairas** [reply]
No worries! One thing I overlooked: the board physically obstructs some additional pins on the GPIO connector. You can still access them e.g. by adding another header on the proto board area and connecting the pins with wire wraps.

**Reply by psm12** [reply]
Thanks,
The engine hat has all the circuit protection that I'm looking for and therefore once I've got to grips with SensESP, PlatformIO, C++........ I'll be purchasing.
